plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -151,16 +170,16 @@ public void registerDataFile(DataFile dataFile, PartitionSpec partitionSpec)
                     .map(PartitionField::sourceId)
                     .collect(toImmutableSet());
             Map<Integer, Optional<String>> partitionValues = getPartitionKeys(dataFile.partition(), partitionSpec);
-            Map<Integer, Object> lowerBounds = convertBounds(idToTypeMapping, dataFile.lowerBounds());
-            Map<Integer, Object> upperBounds = convertBounds(idToTypeMapping, dataFile.upperBounds());
+            Optional<Map<Integer, Object>> lowerBounds = convertBounds(dataFile.lowerBounds());
+            Optional<Map<Integer, Object>> upperBounds = convertBounds(dataFile.upperBounds());
             for (Types.NestedField column : columns) {
                 int id = column.fieldId();
-                io.trino.spi.type.Type trinoType = toTrinoType(column.type(), typeManager);
+                io.trino.spi.type.Type trinoType = icebergToTrinoType.get(column.type());
                 if (identityPartitionFieldIds.contains(id)) {
+                    verify(partitionValues.containsKey(id), "Unable to find value for partition column with field id " + id);
```

Comment:
This is what breaks the schema evolution.
I notice that you didn't have this check and deemed OK.

Maybe we want `if (identityPartitionFieldIds.contains(id) && partitionValues.containsKey(id))`, so that we still try to take min/max from file stats?

Actually, is `identityPartitionFieldIds.contains(id)` important?
if `partitionValues.containsKey(id)` should be enough. The _current table partitioning_ is not important when calculating the stats.

Reply:
> Actually, is identityPartitionFieldIds.contains(id) important?

I was using that to proxy checking if the partition has a transform. If the partitioning is on `hour(ts)` we can't use the partition information to calculate `max(ts)`, but you're right we can't use the current partitioning it needs to be the spec for that file.